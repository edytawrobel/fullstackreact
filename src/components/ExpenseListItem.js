import React from 'react';
import { connect } from 'react-redux';

const ExpenseListItem = ({ description, amount, createdAt }) => (
    <div>
        <h4>Description: { description }</h4>
        <p>Amount: { amount }</p>
        <p>Created At: { createdAt }</p>
    </div>
);

export default ExpenseListItem;
