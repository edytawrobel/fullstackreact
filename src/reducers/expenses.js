const expensesReducerDefaultState = [];

export default ( state = expensesReducerDefaultState, action ) => {
    switch (action.type) {
        case 'ADD_EXPENSE' :
           return [
               ...state,   // spread operator, all expenses
               action.expense
           ];
        case 'REMOVE_EXPENSE' :
            // return state.filter((expense) => {
            //     expense !== action.id // expense was a match and needs to get filtered out
            // });

            return state.filter(( { id }) => id !== action.id);
        case 'EDIT_EXPENSE' :
           return state.map((expense) => {
              if (expense.id === action.id) {
                return {
                    ...expense,
                    ...action.updates
                }
              } else {
                  return expense
              }
           });
        default:
            return state;
    };
}; 

