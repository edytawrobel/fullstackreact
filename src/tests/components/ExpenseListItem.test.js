import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem', () => {
    //const wrapper = shallow(<ExpenseListItem expense={ expenses[1].id} description= {expenses[1].description} amount={expenses[1].amount} createdAt={expenses[1].createdAt} />);
    const wrapper = shallow(<ExpenseListItem { ...expenses[0] } />);
    expect(wrapper).toMatchSnapshot();
}); 
