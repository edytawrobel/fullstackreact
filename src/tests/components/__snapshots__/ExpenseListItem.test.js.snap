// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
     
    <h4>
      Description: 
      Gum
    </h4>
     
  </Link>
  <p>
    Amount: 
    195
  </p>
  <p>
    Created At: 
    0
  </p>
</div>
`;
